pytest/tests/mocknet/load_test_spoon.py

```diff
@@ -110,192 +110,288 @@ def check_slow_blocks(initial_metrics, final_metrics):
         )
-        mocknet.start_nodes(all_nodes, upgrade_schedule)
+        mocknet.start_nodes(self.all_nodes, self.upgrade_schedule)
         time.sleep(60)
 
-    mocknet.wait_all_nodes_up(all_nodes)
+        logger.info(f'Restarting -- done')
 
-    archival_node = rpc_nodes[0]
-    logger.info(f'Archival node: {archival_node.instance_name}')
-    initial_validator_accounts = mocknet.list_validators(archival_node)
-    logger.info(f'initial_validator_accounts: {initial_validator_accounts}')
-    test_passed = True
-
-    script = 'load_test_spoon_helper.py'
-    # TODO: Get these constants from load_test_spoon_helper:
-    # deploy_time = load_test_spoon_helper.CONTRACT_DEPLOY_TIME
-    # test_timeout = load_test_spoon_helper.TEST_TIMEOUT
-    deploy_time = 10 * mocknet.NUM_ACCOUNTS
-    test_timeout = 12 * 60 * 60
-
-    if not args.skip_load:
+    def __start_load_test_helpers(self):
         logger.info('Starting transaction spamming scripts.')
+        script = 'load_test_spoon_helper.py'
         mocknet.start_load_test_helpers(
-            validator_nodes,
             script,
-            rpc_nodes,
-            num_nodes,
-            max_tps,
+            self.validator_nodes,
+            self.rpc_nodes,
+            self.max_tps,
+            self.test_timeout,
+            self.contract_deploy_time,
         )
+        logger.info('Starting transaction spamming scripts -- done.')
+
+    def __wait_to_complete(self):
+        full_test_duration = self.test_timeout + self.contract_deploy_time
+
+        msg = 'Waiting for the loadtest to complete'
+        logger.info(f'{msg}: {full_test_duration}s')
 
-        initial_epoch_height = mocknet.get_epoch_height(rpc_nodes, -1)
+        initial_epoch_height = mocknet.get_epoch_height(self.rpc_nodes, -1)
         logger.info(f'initial_epoch_height: {initial_epoch_height}')
         assert initial_epoch_height >= 0
-        initial_metrics = mocknet.get_metrics(archival_node)
-        start_time = time.monotonic()
-        logger.info(
-            f'Waiting for contracts to be deployed for {deploy_time} seconds.')
+
         prev_epoch_height = initial_epoch_height
-        EPOCH_HEIGHT_CHECK_DELAY = 30
-        while time.monotonic() - start_time < deploy_time:
+        start_time = time.monotonic()
+        while True:
+            neard_running = mocknet.is_binary_running_all_nodes(
+                'neard',
+                self.all_nodes,
+            )
+            helper_running = mocknet.is_binary_running_all_nodes(
+                'load_test_spoon_helper.py',
+                self.validator_nodes,
+            )
+
+            logger.info(
+                f'neard is running on {neard_running.count(True)}/{len(neard_running)} nodes'
+            )
+            logger.info(
+                f'helper is running on {helper_running.count(True)}/{len(helper_running)} validator nodes'
+            )
+
+            if not all(neard_running):
+                raise Exception(
```

Comment:
Yeah I was thinking about it a bit too. The skip-setup and skip-restart options are a good start. What else is there to address do you think? I can only think of the upgrade schedule. For the upgrade schedule we could either generate it from a seed instead of random or we could save it and read it from a file. The file approach may be a bit nicer because then it can also be adjusted manually.

Reply:
the whole validator set gets re-initialized [here](https://github.com/near/nearcore/blob/de989b8d393dc4f82b0957cfcdbd17f23af7551e/pytest/tests/mocknet/load_test_spoon.py#L221), so you need to modify that if you want to resume it. Maybe it will actually work anyway, but I'm not 100% sure